pkg/config/config.go

```diff
@@ -48,6 +64,13 @@ type ComponentSettings struct {
 
 // LocalConfig holds all the config relavent to a specific Component.
 type LocalConfig struct {
+	metav1.TypeMeta   `yaml:",inline"`
```

Comment:
do we even need `TypeMeta` here? It is the internal thing that shouldn't be never touched outside this package

Reply:
we shouldn't diverge how proxyconfig and this config looks, lets keep them same

Reply:
why?

Reply:
so 2 approaches -
- we remove this typemeta as the proxylocalconfig already has it
- we make this typemeta unexported

Reply:
@kadel which one do you think works best?

Reply:
done, took 2nd approach